ml-agents-envs/mlagents/envs/environment.py

```diff
@@ -627,6 +620,17 @@ def _get_state(self, output: UnityRLOutputProto) -> AllBrainInfo:
             )
         return _data
 
+    def _update_brain_parameters(
+        self, init_output: UnityRLInitializationOutputProto
+    ) -> None:
+        if init_output is not None:
```

Comment:
If `init_output` can actually be None, the type should be `Optional[UnityRLInitializationOutputProto]`

Reply:
If it is none, nothing happens? Is there a better way to do this? If I make it optional, why would anyone call `self._update_brain_parameters()` since nothing would happen?

Reply:
`Optional[X]` just indicates that it can either be an instance of X, or None (it's shorthand for `Union[X, None]`).

It's OK to leave for now, I need to do another pass on typechecking so that it actually understands what the proto types are.